tasks/shell_crmsh/crm-cib-resource-primitive.yml

```diff
@@ -47,5 +49,7 @@
           {% endif %}
         {% endfor %}
       {% endif %}
+  # crm can get stuck if it encounters error and expects prompt response.
+  timeout: 60
```

Comment:
What is this?   Where is the behavior of the `timeout` keyword documented?

Reply:
@richm Timeout is ansible task parameter. I have added it because crmsh gets stuck if it encounters error and asks for user input. This is handled with expect in constraint tasks, while keeping resource tasks with timeout.

Reply:
ok, I see - https://docs.ansible.com/ansible/latest/reference_appendices/config.html#task-timeout
"Version Added:   2.10"

Note that for RedHat os_family, we still have to support Ansible 2.9, and still have many customers using 2.9, until late 2024 when 2.9 support will be ended.  Not sure if you have the same restrictions.

Reply:
@richm I was doing testing on my execution node with 2.16 so I was not checking version of it.
It seemed to fulfill task of making sure it does not hang forever. Only other way I see to handle it would be using async and then async_status instead of it.

Do you know of other way to ensure that command/shell does not get stuck like this? I dont see any option with crmsh to skip or auto-decline user inputs without using command/shell with echo pipe.

Reply:
> @richm I was doing testing on my execution node with 2.16 so I was not checking version of it. It seemed to fulfill task of making sure it does not hang forever. Only other way I see to handle it would be using async and then async_status instead of it.
> 
> Do you know of other way to ensure that command/shell does not get stuck like this? I dont see any option with crmsh to skip or auto-decline user inputs without using command/shell with echo pipe.

I would say - use `timeout:` unless you are sure that you will have some customers using Ansible 2.9
If you must support Ansible 2.9, then you will need to use async

Reply:
@richm I have removed timeout and replaced it with expect module with response y/n and ERROR.